cmd/fs-v1.go

```diff
@@ -110,17 +178,25 @@ func newFSObjectLayer(fsPath string) (ObjectLayer, error) {
 
 	// Load `format.json`.
 	format, err := loadFormatFS(fsPath)
-	if err != nil && err != errUnformattedDisk {
+	if err != nil && err != errUnformattedDisk && err != errFSFormatOld {
 		return nil, fmt.Errorf("Unable to load 'format.json', %s", err)
 	}
 
 	// If the `format.json` doesn't exist create one.
-	if err == errUnformattedDisk {
+	if err == errUnformattedDisk || err == errFSFormatOld {
 		fsFormatPath := pathJoin(fsPath, minioMetaBucket, fsFormatJSONFile)
+
 		// Initialize format.json, if already exists overwrite it.
-		if serr := saveFormatFS(fsFormatPath, newFSFormatV1()); serr != nil {
+		if serr := saveFormatFS(fsFormatPath, newFSFormatV2()); serr != nil {
```

Comment:
saveFormatFS() should be called after migrateFSObjectMetadataDir() and when this latter is successful.